cloudinit/settings.py

```diff
@@ -31,11 +31,15 @@
     'datasource_list': [
         'NoCloud',
         'ConfigDrive',
+        'Azure',
         'AltCloud',
         'OVF',
         'MAAS',
+        'GCE',
+        'ExtraConfig',
         'Ec2',
         'CloudStack',
+        'SoftLayer',
```

Comment:
Move Softlayer ahead of ec2... normally ec2 is last on the list because theres a few minute delay retrying the metadata service